src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useState, useRef, useEffect } from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  onUpdateTodo: (id: number, parameter: Partial<Todo>) => void;
+  onDeleteTodo: (id: number) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onUpdateTodo,
+  onDeleteTodo,
+}) => {
+  const {
+    id,
+    title,
+    completed,
+  } = todo;
+
+  const [isInEditMode, setIsInEditMode] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isInEditMode]);
+
+  const handleTitleChange = () => {
+    if (!newTitle.trim()) {
+      onDeleteTodo(id);
+      setIsInEditMode(false);
+    }
+
+    if (newTitle.trim() === title) {
+      setIsInEditMode(false);
+    }
+
+    onUpdateTodo(id, { title: newTitle });
+    setIsInEditMode(false);
+  };
+
+  const handleTitleReset = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setIsInEditMode(false);
+    }
+  };
+
+  let expired: number | null;
```

Comment:
Consider using 'useState' or 'useRef' to manage 'expired' variable state instead of declaring it directly in the component.

Reply:
Done.